sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ObscuringErrorHandlerTest.java

```diff
@@ -0,0 +1,123 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ContextConfiguration;
+
+@SpringBootTest(
+    classes = SecurityTestApp.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ContextConfiguration(initializers = DisableSdaAuthInitializer.class)
+class ObscuringErrorHandlerTest {
+
+  @LocalServerPort private int port;
+
+  @Autowired private TestRestTemplate client;
+
+  @Test
+  void shouldTransformToStandardErrors() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<ApiError> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/errorEntity",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getBody)
+        .extracting(ApiError::getTitle)
+        .isEqualTo("HTTP Error 500 occurred.");
+  }
+
+  @Test
+  void shouldNotTransformApiErrors() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<ApiError> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/apiError",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getTitle())
+        .containsExactly(HttpStatus.NOT_IMPLEMENTED, "This method is not implemented yet.");
+  }
+
+  @Test
+  void shouldMapExceptions() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<ApiError> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/forcedError",
+            HttpMethod.GET,
+            new HttpEntity<>(headers),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getTitle())
+        .containsExactly(HttpStatus.INTERNAL_SERVER_ERROR, "An exception occurred.");
+  }
+
+  @Test
+  void shouldGetResponseEntityWithBody() {
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    ResponseEntity<TestResource> exchange =
+        client.exchange(
+            getServerBaseUrl() + "/api/response",
+            HttpMethod.GET,
```

Comment:
GET returns http status CREATED? 🤔

Reply:
I assume, @saadmar wanted to cover as many status codes as possible.

Reply:
@saadmar : Does it break any test if we change the status code to 200 as far as you remember?

Reply:
I think it does not, and it is probably just a mistake from my side

Reply:
Fixed [here](https://github.com/SDA-SE/sda-spring-boot-commons/pull/162/files#diff-fea3f02b196814b057202c2233d875dfa5e423ebb1424b538d442fef7574c4daR55).